### PR TITLE
Close CodeQL open-redirect alert in 413 handler

### PR DESCRIPTION
**Pull request by annus3**
### Description
Follow-up to #77 — closes CodeQL alert #15 (`py/url-redirection`).

The merged version of PR #77 included an earlier attempt at the open-redirect fix that used a `_safe_referrer()` helper with a netloc-vs-request.host comparison. CodeQL's `py/url-redirection` rule only recognises three sanitizer patterns (strict allowlist, relative-URL-only via empty netloc *and* scheme, or Django's `url_has_allowed_host_and_scheme`) netloc comparison against `request.host` isn't one of them, so the alert kept firing.

### Fix
Remove `_safe_referrer()` entirely; always redirect to `url_for("main.index")` after a 413. No user-controlled value flows into `redirect()` anymore, so the dataflow CodeQL flags doesn't exist.

UX impact: after a >100 MB upload (the 413 threshold), the user lands on the home page instead of being bounced back to the upload form. The flash message tells them what happened. Acceptable trade-off for an edge case that's also the standard phishing primitive.

### Testing
- `pytest` — 37 tests passing
- `test_413_always_redirects_to_home_ignoring_referrer` asserts an attacker-controlled `Referer` header is ignored

CodeQL alert #15 should auto-close on next scan after merge.

### CODE REVIEW

File: Frontend/tests/test_error_handlers.py [hunk 1]
```diff
@@ -53,9 +55,12 @@ def test_413_does_not_open_redirect_to_external_referrer(client, fake_backend, a
...
     assert resp.status_code == 303
-    # Redirect target must be our home page, NOT the attacker's domain.
-    assert "attacker.example.com" not in resp.headers["Location"]
-    assert resp.headers["Location"].endswith("/")
+    location = resp.headers["Location"]
+    # Target must be our home page, never the attacker's domain.
+    assert "attacker.example.com" not in location
+    # Location should point at the index route ("/") — either absolute on our
+    # test host or relative depending on Werkzeug version.
+    assert location.endswith("/") or location.rstrip("/").endswith("localhost")
```
[the remark this answers is not in the document]
Comment (annus3) [reply Frontend/tests/test_error_handlers.py]: Good catch tightened in 1885739. Test now asserts `location in {"/", "http://localhost/"}` so any unintended change to the 413 redirect target will fail the test instead of silently passing.

File: Frontend/web/__init__.py [hunk 2]
```diff
@@ -82,8 +64,11 @@ def request_too_large(_exc):
             "danger",
         )
         # 303 forces a GET on the redirect, so the browser doesn't try to re-POST.
-        # `_safe_referrer()` blocks open-redirects via the Referer header.
-        return redirect(_safe_referrer() or url_for("main.index")), 303
+        # Always redirect to home — never to a user-controlled value (e.g.
+        # request.referrer) — to eliminate the open-redirect risk (CWE-601).
+        # CodeQL's py/url-redirection rule only recognises hardcoded targets
+        # or strict allowlists as sanitizers, not netloc/host comparisons.
```
[the remark this answers is not in the document]
Comment (annus3) [reply Frontend/web/__init__.py]: fixed in bbef1eb. Test now derives the expected target from `url_for("main.index")` inside `app.test_request_context()` and accepts both the relative and `_external=True` variants. The test is now decoupled from the literal `/` path and will stay correct if the home route ever moves or APPLICATION_ROOT is set.
